src/libs/pulumi-cli.ts

```diff
@@ -12,6 +12,25 @@ export async function isAvailable(): Promise<boolean> {
   return res.stderr != '' && !res.success ? false : res.success;
 }
 
+export async function getVersion(): Promise<string | undefined> {
```

Comment:
My internal style-guide is that undefined should be reserved for instances where a variable is not in scope, while null is intended to convey the absence of a defined value.

That's my preference, but other philosophies might be better informed than mine!

```suggestion
export async function getVersion(): Promise<string | null> {
```

Reply:
Internally, the service team follows https://github.com/microsoft/TypeScript/wiki/Coding-guidelines

https://github.com/microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined

undefined is preferred over null

Reply:
Well, judging by the H1s at the top of the document, I'm inclined to disregard advice without any reasoning provided ;D

Reply:
undefined is the norm for typescript here. The optional type `string?` would expand out to `string | undefined`. The TS tooling probably understands that style better.

Reply:
I fully admit that TypeScript is not my strongest language so I let the tooling guide me.  I'm coming from a C# background so I started with a function return type of `Promise<string?>`. The TS compiler complained with the following message:
```
// JSDoc types can only be used inside documentation comments.(8020)
```
During my investigation into this message I ran across this [blog post](https://paulsebastian.codes/why-isnt-string-allowed-as-a-function-return-value-in-typescript) which laid out a pretty good explanation on why `string?` is considered an invalid return type.

I'm happy to follow whatever style guide is decided upon.